### PR TITLE
feat: reformat help text in the style of cargo

### PR DESCRIPTION
**Pull request by joshka**
My motivation here was that when I run cargo generate, I find it difficult to find the options that are useful as there's no real order to the help output. This helps by injecting color, sorting the options into some reasonable sections, and shortening the longest option name.

# Before:

image
image

# After:

image

### CODE REVIEW

File: Cargo.toml
Comment (joshka) [inline Cargo.toml:1]: The formatting / sorting changes are in a separate commit

File: Cargo.toml [hunk 1]
```diff
@@ -9,58 +9,56 @@ edition = "2021"
...
-git2 = { version = "~0.19", features = ["ssh", "https"], default-features = false }
+anstyle = "1.0.7"
+anyhow = "~1.0"
 auth-git2 = "~0.5"
+clap = { version = "~4.5", default-features = true, features = [
+    "derive",
+    "std",
+    "help",
+    "wrap_help",
+] }
```
Comment (joshka) [inline Cargo.toml:20]: This is the only real change in this file (adding wrap help and turning on default features, which activates the color and the usage features)

File: src/args.rs [hunk 2]
```diff
@@ -9,13 +9,47 @@ use serde::Deserialize;
...
+            .valid(VALID)
+            .invalid(INVALID)
+            .error(ERROR)
+    };
+}
+
+mod heading {
+    pub const GIT_PARAMETERS: &str = "Git Parameters";
+    pub const TEMPLATE_SELECTION: &str = "Template Selection";
+    pub const OUTPUT_PARAMETERS: &str = "Output Parameters";
```
Comment (joshka) [inline src/args.rs:41]: I wasn't 100% sure of good names for these headings, and there's a few options that I wasn't entirely sure whether to leave them in the main options section or in the output params (e.g. overwrite, allow-commands)

File: src/args.rs [hunk 3]
```diff
@@ -52,77 +86,78 @@ pub struct GenerateArgs {
...
     /// Enables more verbose output.
     #[arg(long, short, action)]
     pub verbose: bool,
 
-    /// Pass template values through a file
-    /// Values should be in the format `key=value`, one per line
-    #[arg(long, value_parser)]
+    /// Pass template values through a file. Values should be in the format `key=value`, one per
+    /// line
+    #[arg(long="values-file", value_parser, alias="template-values-file", value_name="FILE", help_heading = heading::OUTPUT_PARAMETERS)]
```
Comment (joshka) [inline src/args.rs:103]: Added a shorter name here to make the formatting nicer (and also to make this easier for a user to type). I left the original alias however so any existing scripts etc. can still use that name.

File: Cargo.toml [hunk 4]
```diff
@@ -9,58 +9,51 @@ edition = "2021"
 include = ["src/**/*", "LICENSE-*", "*.md"]
 
 [dependencies]
-clap = { version = "~4.5", features = ["derive", "std", "help"], default-features = false }
-git2 = { version = "~0.19", features = ["ssh", "https"], default-features = false }
+anstyle = "1.0.7"
+anyhow = "~1.0"
 auth-git2 = "~0.5"
+clap = { version = "~4.5", features = ["derive", "wrap_help"] }
```
Comment (joshka) [inline Cargo.toml:15]: This is the only real change in this file (adding wrap help and turning on default features, which activates the color and the usage features).

There's no specific reason in the git / PR history why default features was turned off here for clap. See https://github.com/cargo-generate/cargo-generate/pull/655

File: src/args.rs (hunk 2 above)
Comment (sassman) [reply src/args.rs:41]: for me it sounds reasonable.

File: src/args.rs (hunk 3 above)
Comment (sassman) [reply src/args.rs:103]: thanks for keeping it as a non-breaking change.

**Review by sassman — APPROVED** [review]
Thanks a lot for this huge improvement! I know that several folks already asked for this.

**Comment by sassman** [comment]
this PR addresses parts of #894

File: Cargo.toml (hunk 4 above)
Comment (sassman) [reply Cargo.toml:15]: fair point.

**Comment by joshka** [comment]
Thanks for the quick merge :)

**Pull request merged**